### DISCUSSION TITLE
Option -C doesn't (seem to) work

### DISCUSSION THREAD

**Discussion by tjcreedy** [nextflow-io/nextflow#4452] [Q&A]
## Bug report 

If I run any nextflow command specifying a config file with `-c`, it works fine, but if I switch to `-C` it gives an error. Am I missing something about how the `-C` option should work?

### Expected behavior and actual 

Nextflow should run using only the specified config file and no other config files; instead, it gives an error.

### Steps to reproduce the problem

```
touch test.config
nextflow run hello -c test.config # runs fine
nextflow run hello -C test.config # error
```

### Program output 

```
Unknown option: -C -- Check the available commands and options and syntax with 'help'
```

### Environment 

* Nextflow version: 23.10.0.5889 
* Java version: openjdk 21-internal 2023-09-19
* Operating system: Pop!_OS 22.04 LTS (also tested on Ubuntu 22.04 and 20.04)
* Bash version: 5.1.16(1)

**Comment by bentsherman** [comment]
This is because `-c` and `-C` are top-level options, not `run` options, however `-c` is also secretly supported by the `run` command as a convenience. This should work:

```bash
nextflow -C test.config run hello
```

**Comment by mpiersonsmela** [comment]
I ran into this problem too. This is NOT listed anywhere in the documentation and switching from -c to -C broke a lot of stuff for me until I found this on Github.
